controllers/kokumetricsconfig_controller.go

```diff
@@ -517,33 +513,43 @@ func uploadFiles(r *KokuMetricsConfigReconciler, authConfig *crhchttp.AuthConfig
 	return nil
 }
 
-func collectPromStats(r *KokuMetricsConfigReconciler, kmCfg *kokumetricscfgv1beta1.KokuMetricsConfig, dirCfg *dirconfig.DirectoryConfig) {
-	log := log.WithName("collectPromStats")
+func getTimeRange(r *KokuMetricsConfigReconciler, kmCfg *kokumetricscfgv1beta1.KokuMetricsConfig) (time.Time, time.Time) {
+	start := time.Now().UTC().Truncate(time.Hour).Add(-time.Hour) // start of previous full hour
+	end := start.Add(59*time.Minute + 59*time.Second)
+	if kmCfg.Status.Prometheus.LastQuerySuccessTime.IsZero() && r.InCluster && !r.disablePreviousDataCollection {
+		start = time.Date(start.Year(), start.Month(), 1, 0, 0, 0, 0, start.Location())
```

Comment:
So this will at most gather data back to beginning of the current month?

Would it be better to gather a fixed number of days of data?

What happens if we request data for a date range where there is no data? I believe that scenario is handled in `collectPromStats` just wanted to make sure.

Reply:
The best solution is to gather whatever is available, but we don't know when the data starts. Given that most billing starts at the beginning of a month, that makes it a good place to start.

If there is no data, we just move to the next hour.

Reply:
That makes sense. I hadn't thought about lining up with billing date. I was just thinking "get all the data available".